src/dispatch/tag/recommender.py

```diff
@@ -0,0 +1,171 @@
+
+    correlation_matrix_dict = {}
+
+    for tag_a in unique_tags:
+        correlation_matrix_dict[tag_a] = correlate_with_every_tag(dataframe, tag_a)
+
+    correlated_dataframe = pd.DataFrame(correlation_matrix_dict)
+    correlated_dataframe["index"] = unique_tags
+    return correlated_dataframe.set_index("index")
+
+
+def create_boolean_dataframe(items: List[Any]):
+    """Create a boolean dataframe with tag and item data."""
+    unique_tags = get_unique_tags(items)
+    boolean_df = pd.DataFrame(columns=unique_tags.values())
+
+    data_dict = defaultdict(list)
+    for col in boolean_df:
+        for i in items:
+            tag_ids = [t.id for t in i.tags]
+            data_dict[col].append(col in tag_ids)
+
+    for col in boolean_df:
+        boolean_df[col] = data_dict[col]
+
+    return boolean_df
+
+
+def find_correlations(dataframe, tag):
+    """Find all correlations for the given tag."""
+    # Setup empty list
+    correlations = []
+    columns = []
+
+    # Loop through all column at the row with the tag as its index
+    for i, corr in enumerate(dataframe.loc[tag, :]):
+
+        # Find the column
+        col = dataframe.columns[i]
+
+        # Append the correlation to the list
+        correlations.append(corr)
+        columns.append(col)
+
+    # Create a df out of the lists
+    results_df = pd.DataFrame({"tag": columns, "correlation": correlations})
+
+    return results_df
+
+
+def find_highest_correlations(correlated_dataframe, recommendations):
+    """Find the correlations with the highest relevancy."""
+    # Sort the input df
+    corr_df_sorted = correlated_dataframe.sort_values(by=["correlation"], ascending=False)
+
+    # Extract the relevant correlations
+    corr_df_sliced = corr_df_sorted.iloc[1 : recommendations + 1]  # noqa
+
+    return corr_df_sliced
+
+
+def get_recommendations(db_session: SessionLocal, tag_ids: List[str], model_name: str, recommendations: int = 5):
+    """Get recommendations based on current tag."""
+    try:
+        correlation_dataframe = load_model(model_name)
+    except FileNotFoundError:
+        log.warning(f"No model file found. ModelName: {model_name}")
+        return []
+
+    recommendations_dataframe = pd.DataFrame()
+    for tag_id in tag_ids:
+        correlated_dataframe = find_correlations(correlation_dataframe, tag_id)
+        recommendations_dataframe = pd.concat(
+            [
+                recommendations_dataframe,
+                find_highest_correlations(correlated_dataframe, recommendations),
+            ],
+            ignore_index=True,
+        )
+
+    # convert back to tag objects
+    tags = []
+    for t in recommendations_dataframe[:recommendations]:
+        tags.append(tag_service.get(db_session=db_session, tag_id=t))
+
+    log.debug(f"Making tag recommendation. TagIds: {','.join([t.name for t in tags])} ModelName: {model_name} Recomendations: ", list(recommendations_dataframe["tag"][:recommendations]))
```

Comment:
Is this going to work? Shouldn't `list(recommendations_dataframe["tag"][:recommendations])` go between `{}` and inside the `""`?